pkl-core/src/main/java/org/pkl/core/runtime/VmMapping.java

```diff
@@ -127,6 +127,20 @@ public Map<Object, Object> toMap() {
 
     return properties;
   }
+  
+  public VmDynamic toDynamic() {
+    var keys = getAllKeys();
+    var builder = new VmObjectBuilder(keys.getLength());
+    for (var key: keys) {
+      var member = VmUtils.readMember(this, key);
```

Comment:
Instead of forcing and copying each value, the Dynamic could delegate to the Mapping via `addEntry(Object key, SharedMemberNode valueNode)`.
A similar overload could be added for `addProperty` (I didn't need it at the time).